streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java

```diff
@@ -172,6 +174,11 @@ public void addStateStore(String stateStoreName) {
         public ProcessorNode build() {
             return new ProcessorNode<>(name, supplier.get(), stateStoreNames);
         }
+
+        @Override
+        TopologyDescription.Processor describe() {
```

Comment:
It's just to have stronger type safety. We know, that we only return `Processor` but nothing else. I don't see any disadvantage in having a strong type.